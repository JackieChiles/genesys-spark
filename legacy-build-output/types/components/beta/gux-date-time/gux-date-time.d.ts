import { JSX } from '../../../stencil-public-runtime';
import { GuxDateTimeFormat } from '../../../i18n/DateTimeFormatter';
export declare class GuxDateTime {
  private formatter;
  /**
   * Reference to the host element.
   */
  root: HTMLElement;
  /**
   * The ISO string representation of the datetime to format
   */
  datetime: string;
  /**
   * Format option type
   */
  format: GuxDateTimeFormat;
  componentWillLoad(): void;
  render(): JSX.Element;
}
