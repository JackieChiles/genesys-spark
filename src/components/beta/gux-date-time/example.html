<h1>gux-date-time-beta</h1>

<h2>Date and Time</h2>

<h3>Short</h3>
<gux-date-time-beta id="time-1" format="short"></gux-date-time-beta>

<h3>Medium</h3>
<gux-date-time-beta id="time-2" format="medium"></gux-date-time-beta>

<h3>Long</h3>
<gux-date-time-beta id="time-3" format="long"></gux-date-time-beta>

<h3>Full</h3>
<gux-date-time-beta id="time-4" format="full"></gux-date-time-beta>

<style
  onload="(function() {
    const time1 = document.querySelector('#time-1');
    const time2 = document.querySelector('#time-2');
    const time3 = document.querySelector('#time-3');
    const time4 = document.querySelector('#time-4');

    const date = new Date();

    time1.date = date;
    time2.date = date;
    time3.date = date;
    time4.date = date;
  })()"
>
  .not-used {
    -custom-noop: noop;
  }
</style>
