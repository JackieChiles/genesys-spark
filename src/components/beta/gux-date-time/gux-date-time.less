@import (reference) '../../../style/spacing.less';

.gux-date-time {
    display: flex;
    padding: @spacing-xs @spacing-xs;
}