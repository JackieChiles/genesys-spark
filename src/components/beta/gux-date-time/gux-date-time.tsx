import { Component, h, Host, Prop, Element } from '@stencil/core';
import { trackComponent } from '../../../usage-tracking';
import { DateTimeFormatter, GuxDateTimeFormat } from '../../../i18n/DateTimeFormatter';
import { getDesiredLocale } from '../../../i18n/index';

@Component({
  styleUrl: 'gux-date-time.less',
  tag: 'gux-date-time-beta',
  shadow: true
})
export class GuxDateTime {
  private formatter : DateTimeFormatter;

  /**
   * Reference to the host element.
   */
  @Element()
  root: HTMLElement;

  /**
   * The ISO string representation of the datetime to format 
   */
  @Prop()
  datetime: string = new Date().toISOString();

  /**
   * Format option type 
   */
  @Prop()
  format: GuxDateTimeFormat = 'full';

  componentWillLoad(): void {
    trackComponent(this.root);
    this.formatter = new DateTimeFormatter(getDesiredLocale(this.root));
  }

  render() {
    return (<Host class="gux-date-time">{ this.formatter.formatDateTime(new Date(this.datetime), this.format) }</Host>);
  }
}
