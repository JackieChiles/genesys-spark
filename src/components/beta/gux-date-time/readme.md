# gux-date-time-beta



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description               | Type                                      | Default      |
| -------- | --------- | ------------------------- | ----------------------------------------- | ------------ |
| `date`   | --        | The Date object to format | `Date`                                    | `new Date()` |
| `format` | `format`  | Format option type             | `"full" \| "long" \| "medium" \| "short"` | `'full'`     |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
