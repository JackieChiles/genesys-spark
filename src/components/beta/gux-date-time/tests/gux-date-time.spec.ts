import { newSpecPage } from '@stencil/core/testing';
import { GuxDateTime } from '../gux-date-time';

const components = [GuxDateTime];
const language = 'en';
const date = new Date(2022, 6, 7, 9, 35, 30, 100).valueOf();

describe('#render different datetime formats eg(short,medium,full,long)', () => {
    [
      `<gux-date-time-beta locale="en" date=${date} format="short"></gux-date-time-beta>`,
      `<gux-date-time-beta locale="en" date=${date} format="medium"></gux-date-time-beta>`,
      `<gux-date-time-beta locale="en" date=${date} format="long"></gux-date-time-beta>`,
      `<gux-date-time-beta locale="en" date=${date} format="full"></gux-date-time-beta>`
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {

        const page = await newSpecPage({ components, html, language });
  
        expect(page.root).toMatchSnapshot();
      });
    });
  });
