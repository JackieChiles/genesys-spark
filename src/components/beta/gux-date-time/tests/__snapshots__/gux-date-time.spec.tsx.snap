// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render different datetime formats eg(short,medium,full,long) should render component as expected (1) 1`] = `
<gux-date-time-beta>
  <mock:shadow-root>
    <span>
      7/7/22, 9:35 AM
    </span>
  </mock:shadow-root>
</gux-date-time-beta>
`;

exports[`#render different datetime formats eg(short,medium,full,long) should render component as expected (2) 1`] = `
<gux-date-time-beta>
  <mock:shadow-root>
    <span>
      Jul 7, 2022, 9:35:30 AM
    </span>
  </mock:shadow-root>
</gux-date-time-beta>
`;

exports[`#render different datetime formats eg(short,medium,full,long) should render component as expected (3) 1`] = `
<gux-date-time-beta>
  <mock:shadow-root>
    <span>
      Thursday, July 7, 2022 at 9:35:30 AM GMT-04:00
    </span>
  </mock:shadow-root>
</gux-date-time-beta>
`;

exports[`#render different datetime formats eg(short,medium,full,long) should render component as expected (4) 1`] = `
<gux-date-time-beta>
  <mock:shadow-root>
    <span>
      July 7, 2022 at 9:35:30 AM GMT-4
    </span>
  </mock:shadow-root>
</gux-date-time-beta>
`;
